PyMovieDb/imdb.py

```diff
@@ -79,73 +80,76 @@ def search(self, name, year=None, tv=False, person=False):
+                            })
+            for item in results["props"]["pageProps"]["nameResults"]["results"]:
+                output.append({
+                        "id": item['id'],
+                        "name": item['displayNameText'],
+                        "url": f"{self.baseURL}/name/{item['id']}",
+                        "poster": item.get('avatarImageModel', {}).get('url', ''),
+                        "type": item["knownForJobCategory"]
+                        })
+
+        self.search_results = {'result_count': len(output), 'results': output}
+        return self.search_results
 
     # ..............................methods to get a movie/web-series/tv info..............................
     def get(self, url):
         """
          @description:- helps to get a file's complete info (used by get_by_name() & get_by_id() )
          @parameter:- <str:url>, url of the file/movie/tv-series.
-         @returns:- File/movie/TV info as JSON string.
+         @returns:- File/movie/TV info as JSON.
         """
         try:
             response = self.session.get(url)
-            result = response.html.xpath("//script[@type='application/ld+json']")[0].text
-            result = ''.join(result.splitlines())  # removing newlines
-            result = f"""{result}"""
+            soup = BeautifulSoup(response.content,'html.parser')
+            result = soup.find('script', type='application/ld+json').text
             # print(result)
-        except IndexError:
-            return self.NA
-        try:
-            # converting json string into dict
+            # converting JSON object into dict
             result = json.loads(result)
-        except json.decoder.JSONDecodeError as e:
-            # sometimes json is invalid as 'description' contains inverted commas or other html escape chars
-            try:
-                to_parse = ImdbParser(result)
-                # removing trailer & description schema from json string
-                parsed = to_parse.remove_trailer
-                parsed = to_parse.remove_description
-                # print(parsed)
-                result = json.loads(parsed)
-            except json.decoder.JSONDecodeError as e:
-                try:
-                    # removing reviewBody from json string
-                    parsed = to_parse.remove_review_body
-                    result = json.loads(parsed)
-                except json.decoder.JSONDecodeError as e:
-                    # invalid char(s) is/are not in description/trailer/reviewBody schema
-                    return self.NA
+            result = ImdbParser(result).unescape_json_values
+        except (json.decoder.JSONDecodeError, IndexError) as e:
+            return self.NA
 
         output = {
             "type": result.get('@type'),
             "name": result.get('name'),
+            "alternateName": result.get('alternateName', None),
             "url": self.baseURL + result.get('url'),
```

Comment:
maybe it's not the case all the time, but I found out that `result.get('url')` already has imdb.com inside.
```suggestion
            "url": result.get('url'),
```